projects/media-viewer/src/lib/shared/viewer.events.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+import {Subject} from 'rxjs';
+
+
+// Viewer Custom-Event Types
+export interface TextSelectionRectangle {
+  x: number;
+  y: number;
+  height: number;
+  width: number;
+}
+export type TextSelected = TextSelectionRectangle[];
+
+
+@Injectable({providedIn: 'root'})
+export class ViewerEventsService {
+  // Register Observable Subject Events relevant to the Viewers
+  textSelected = new Subject<TextSelected>();
+  constructor() {}
+
+  // Function to inform Observers that text has been selected in the viewer
+  public onTextSelection(selectionData: TextSelected): void {
```

Comment:
I'm  not sure about this file...  the service approach makes sense for `ToolbarEventsService` as it's encapsulating the state like a repository would. This feels a bit different though because it's encapsulating what should be an `@Output()` event.

Reply:
I see why you are suggesting that, however, I see a need to act on Viewer Events from other components… e.g the Toolbar, or alternatively if we split the comments out, that will too need to become viewer event aware…

Reply:
okay, I'd like to know what @benouaer thinks